### ISSUE TITLE
About Data Layer

### ISSUE THREAD

**Issue by liuxunfu** [xinghaochen/Pose-REN#11]
Hi,thanks for sharing your project, I have two questions here：
    1 Is the training data layer defined in python?
    2 In the training label of icvl data set, there are 48 floating-point numbers. How to realize multi-label training?

**Comment by xinghaochen** [comment]
Hi，
1. You can implement the data layer in either Python or C++, we used C++.
2. The pose label is a vector of size (48, ). Therefore, just feed the annotations as a blob of size (batch_size, 48) into the model.

**Comment by liuxunfu** [comment]
> Hi，
> 
>     1. You can implement the data layer in either Python or C++, we used C++.
> 
>     2. The pose label is a vector of size (48, ). Therefore, just feed the annotations as a blob of size (batch_size, 48) into the model.

Ok, thank you. By the way, how is the following achieved?
    "We extracted a fix-sized cube from the input depth image. The center of the cube was determined by calculating the centroid of mass of the hand region. The extracted cube was then resized into a patch with size of 96 × 96 and the depth values within it were normalized into [−1, 1]."

**Comment by xinghaochen** [comment]
Please see [src/utils/util.py](https://github.com/xinghaochen/Pose-REN/blob/HEAD/src/utils/util.py#L244) and [src/utils/model_pose_ren.py](https://github.com/xinghaochen/Pose-REN/blob/HEAD/src/utils/model_pose_ren.py#L120).

**Comment by liuxunfu** [comment]
> Please see [src/utils/util.py](https://github.com/xinghaochen/Pose-REN/blob/HEAD/src/utils/util.py#L244) and [src/utils/model_pose_ren.py](https://github.com/xinghaochen/Pose-REN/blob/HEAD/src/utils/model_pose_ren.py#L120).

Ok, thank you very much.
